### ISSUE TITLE
Exception raised when upgraded to ruby 3.3.1 and deployed to Heroku

### ISSUE THREAD

**Issue by obromios** [Shopify/bootsnap#484]
### I made sure the issue is in bootsnap
When I commented bootstrap out of the Gemfile and commented out require "bootsnap/setup" from `boot.rb` then was able to deploy successfully.

### Steps to reproduce

Upgrade to ruby 3.3.1 and `bundle install`, and commit `Gemfile` and `Gemfile.lock`
Then `git push production master`
### Expected behavior

It should deploy to production.
### Actual behavior

Saw following exception

```
remote: /tmp/codon/tmp/buildpacks/xxx/lib/language_pack/helpers/rake_runner.rb:100:in `load_rake_tasks!': Could not detect rake tasks (LanguagePack::Helpers::RakeRunner::CannotLoadRakefileError)
remote: ensure you can run `$ bundle exec rake -P` against your app
remote: and using the production group of your Gemfile.
remote: /app/vendor/bundle/ruby/3.3.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30: warning: /app/vendor/ruby-3.3.1/lib/ruby/3.3.0/base64.rb was loaded from the standard library, but will no longer be part of the default gems since Ruby 3.4.0. Add base64 to your Gemfile or gemspec. Also contact author of activesupport-7.0.8.1 to add base64 into its gemspec.
remote: /app/vendor/bundle/ruby/3.3.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30: warning: /app/vendor/ruby-3.3.1/lib/ruby/3.3.0/bigdecimal.rb was loaded from the standard library, but will no longer be part of the default gems since Ruby 3.4.0. Add bigdecimal to your Gemfile or gemspec. Also contact author of activesupport-7.0.8.1 to add bigdecimal into its gemspec.
remote: /app/vendor/bundle/ruby/3.3.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30: warning: /app/vendor/ruby-3.3.1/lib/ruby/3.3.0/mutex_m.rb was loaded from the standard library, but will no longer be part of the default gems since Ruby 3.4.0. Add mutex_m to your Gemfile or gemspec. Also contact author of activesupport-7.0.8.1 to add mutex_m into its gemspec.
remote: /app/vendor/ruby-3.3.1/lib/ruby/3.3.0/bundled_gems.rb:72: warning: /app/vendor/ruby-3.3.1/lib/ruby/3.3.0/csv.rb was loaded from the standard library, but will no longer be part of the default gems since Ruby 3.4.0. Add csv to your Gemfile or gemspec. Also contact author of blazer-2.6.5 to add csv into its gemspec.
remote: rake aborted!
remote: ArgumentError: comparison of String with nil failed (ArgumentError)
remote: 
remote:     msg = " #{RUBY_VERSION < SINCE[gem] ? "will no longer be" : "is not"} part of the default gems since Ruby #{SINCE[gem]}."
remote:                              ^^^^^^^^^^
remote: /app/vendor/ruby-3.3.1/lib/ruby/3.3.0/bundled_gems.rb:130:in `<'
remote: /app/vendor/ruby-3.3.1/lib/ruby/3.3.0/bundled_gems.rb:130:in `build_message'
remote: /app/vendor/ruby-3.3.1/lib/ruby/3.3.0/bundled_gems.rb:126:in `warning?'
remote: /app/vendor/ruby-3.3.1/lib/ruby/3.3.0/bundled_gems.rb:71:in `block (2 levels) in replace_require'
remote: /app/vendor/bundle/ruby/3.3.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
```
### System configuration
**Bootsnap version**:
`1.18.3`
**Ruby version**:
`3.3.1`
**Rails version**: 
`7.0.8.a`

**Comment by casperisfine** [comment]
> Unfortunately adding mutex_m to the gemfile did not resolve this for me. Do you have some additional context or advice that could help work around this?

After reformating your issue, it now seems clear `mutex_m` is just the first of three issues:

```
remote: /app/vendor/bundle/ruby/3.3.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30: warning: /app/vendor/ruby-3.3.1/lib/ruby/3.3.0/base64.rb was loaded from the standard library, but will no longer be part of the default gems since Ruby 3.4.0. Add base64 to your Gemfile or gemspec. Also contact author of activesupport-7.0.8.1 to add base64 into its gemspec.
remote: /app/vendor/bundle/ruby/3.3.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30: warning: /app/vendor/ruby-3.3.1/lib/ruby/3.3.0/bigdecimal.rb was loaded from the standard library, but will no longer be part of the default gems since Ruby 3.4.0. Add bigdecimal to your Gemfile or gemspec. Also contact author of activesupport-7.0.8.1 to add bigdecimal into its gemspec.
remote: /app/vendor/bundle/ruby/3.3.0/gems/bootsnap-1.18.3/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30: warning: /app/vendor/ruby-3.3.1/lib/ruby/3.3.0/mutex_m.rb was loaded from the standard library, but will no longer be part of the default gems since Ruby 3.4.0. Add mutex_m to your Gemfile or gemspec. Also contact author of activesupport-7.0.8.1 to add mutex_m into its gemspec.
```

So you not only need `mutex_m`, but also `bigdecimal` and `base64`. I'll see about backporting this fix to Active Support 7.0 branch.

**Comment by casperisfine** [comment]
Actually that was done a while ago already, just not released yet: https://github.com/rails/rails/pull/50546

**Comment by obromios** [comment]
I tried deploying with
gem 'base64'
gem 'bigdecimal'
gem 'mutex_m'
and still had the same problem.

**Comment by obromios** [comment]
This issue is fixed with version 7.0.8.3 of Rails.
